File: typescript/day3/part1.ts
const inputFile = 'input.txt'
const inputData = await Deno.readTextFile(inputFile)

const sum = [...inputData.matchAll(/mul\([0-9]{1,3},[0-9]{1,3}\)/g)]
	.map(mul => {
		const [multiplicand, multiplier] =
			[...mul.toString().matchAll(/\d+/g)].map(factor => parseInt(factor.toString()))
		return multiplicand * multiplier
	})
	.reduce((prev, curr) => prev + curr, 0)

console.log(`sum = ${sum}`)

File: rust/day3/src/part1.rs
use std::fs;

use regex::Regex;


fn main() {
    let input_data = fs::read_to_string("input.txt").expect("Could not read input file.");

	let re = Regex::new(r"mul\((\d{1,3}),(\d{1,3})\)").unwrap();

	let mut sum = 0;

	for (_, [a, b]) in re.captures_iter(&input_data).map(|cap| cap.extract()) {
		sum += a.parse::<i32>().unwrap() * b.parse::<i32>().unwrap();
	}

	println!("{sum}");
}
